src/pkcs11/framework-pkcs15.c

```diff
@@ -5577,22 +5577,22 @@ get_ec_pubkey_params(struct sc_pkcs15_pubkey *key, CK_ATTRIBUTE_PTR attr)
 	switch (key->algorithm) {
 	case SC_ALGORITHM_EDDSA:
 	case SC_ALGORITHM_XEDDSA:
-		r = sc_encode_oid(context, &key->alg_id->oid, &value, (size_t *)&value_len);
+		r = sc_encode_oid(context, &key->alg_id->oid, &value, &value_size);
 		if (r != SC_SUCCESS) {
 			return sc_to_cryptoki_error(r, NULL);
 		}
 
-		attr->ulValueLen = value_len;
+		attr->ulValueLen = (unsigned long)value_size;
 		if (attr->pValue == NULL_PTR) {
 			free(value);
 			return CKR_OK;
 		}
-		if (attr->ulValueLen < value_len) {
+		if (attr->ulValueLen < (unsigned long)value_size) {
```

Comment:
Your commit doesn't introduce this problem, but as it now strikes my eye, we should take a closer look:

1. with `attr->ulValueLen = (unsigned long)value_size;`
2. this will always be false: `attr->ulValueLen < (unsigned long)value_size`

Reply:
Right. Thank you for having a look! Indeed, it is incorrect and caused by my attempt to "optimize" some assignments. The first assignments needs to be in all the exit code paths but not before we know we can fit the output to the allocated buffer to follow pkcs11 semantics similarly as in the other functions. I will submit a fix shortly.